### PR TITLE
feat: configure FTPS state

### PR DESCRIPTION
**Pull request by kov117**
### Checklist

- [x] I've updated both the `azurerm_linux_function_app` and `azurerm_windows_function_app` resources.

Test passed 🎆  

But i have a question for @equinor/terraform-baseline-maintainers, according to this [FTPS](https://learn.microsoft.com/en-us/azure/app-service/deploy-ftp?tabs=portal#enforce-ftps) we can just disable (as i have it by default) both FTP and FTPS if we are not using FTP deployment. But if we want enchanced security, we should allow FTP over TLS/SSL. Do you guys have any thoughs about this ?

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -216,6 +216,12 @@ variable "identity_ids" {
   default     = []
 }
 
+variable "ftps_state" {
+  description = "State of FTP/FTPS service for this Windows Function App."
```
Comment (hknutsen) [inline variables.tf:220]: Phrasing and allowed values (according to [variable descriptions best practices](https://equinor.github.io/terraform-baseline/best-practices/variables-and-outputs/#variable-and-output-descriptions)):

```suggestion
  description = "The state of the FTP / FTPS service for this Function App. Value must be \"AllAllowed\", \"FtpsOnly\" or \"Disabled\"."
```

**Review by hknutsen — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 2]
```diff
@@ -216,6 +216,12 @@ variable "identity_ids" {
...
+  type        = string
+  default     = "Disabled"
+}
```
Comment (hknutsen) [inline variables.tf:223]: Validate allowed values (according to [best practices](https://equinor.github.io/terraform-baseline/best-practices/variables-and-outputs/)):

```suggestion
  default     = "Disabled"
  
  validation {
    condition     = contains(["AllAllowed", "FtpsOnly", "Disabled"], var.ftps_state)
    error_message = "FTPS state must be \"AllAllowed\", \"FtpsOnly\" or \"Disabled\"."
  }
}
```

**Review by hknutsen — CHANGES_REQUESTED** [review]

**Comment by kov117** [comment]
Ran new test and it passed 👍🏼

**Comment by kov117** [comment]
Terraform test passed 👍🏼

**Review by hknutsen — APPROVED** [review]
Looks good 👍

**Pull request merged**